tamr_client/operation.py

```diff
@@ -0,0 +1,195 @@
+"""
+See https://docs.tamr.com/new/reference/the-operation-object
+"""
+from copy import deepcopy
+from dataclasses import dataclass
+from time import sleep, time as now
+from typing import Dict, Optional
+
+import requests
+
+from tamr_client import response
+from tamr_client.instance import Instance
+from tamr_client.session import Session
+from tamr_client.types import JsonDict
+from tamr_client.url import URL
+
+
+class NotFound(Exception):
+    """Raised when referencing an operation that does not exist on the server.
+    """
+
+    pass
+
+
+@dataclass(frozen=True)
+class Operation:
+    """A Tamr operation
+
+    See https://docs.tamr.com/new/reference/the-operation-object
+
+    Args:
+        url
```

Comment:
Are the other attributes intentionally left out?

```suggestion
        url
        id
        type
        status
        description
```

Reply:
I wasn't sure if it was best to include all of these.  Looking at `Dataset`, there are 3 required attributes, `url`, 'name`, `key_attribute_names` and 1 optional attribute `description`.  In the docstring, only `url` and `key_attribute_names` are listed under `Args:`.  Is this intentional, and if so is there a pattern that should be followed, or should every attribute be listed?

Reply:
We should include all user-facing attributes in the docstring, so I think `Dataset` should be fixed.

Reply:
👍  Once this PR is near merging, `unified.commit` should be updated to return an `Operation`.  I'll fix the docstrings in `Dataset` and `UnifiedDataset` then.